File: backend/src/main/java/br/ufpr/oscar/entity/Filme.java
package br.ufpr.oscar.entity;

import jakarta.persistence.*;

@Entity
@Table(name = "filmes")
public class Filme {

    @Id
    private Long id;      

    @Column(nullable = false, length = 200)
    private String nome;

    @Column(nullable = false, length = 100)
    private String genero;

    @Column(nullable = false, length = 500)
    private String foto;

    public Filme() {}

    public Filme(Long id, String nome, String genero, String foto) {
        this.id = id;
        this.nome = nome;
        this.genero = genero;
        this.foto = foto;
    }

    public Long getId()     { return id; }
    public String getNome() { return nome; }
    public String getGenero() { return genero; }
    public String getFoto() { return foto; }
}

File: mobile/app/src/main/java/br/ufpr/oscarmobile/model/Filme.kt
package com.ufpr.oscarmobile.model
 
import com.google.gson.annotations.SerializedName

data class Filme(
    @SerializedName("id")     val id: String,
    @SerializedName("nome")   val nome: String,
    @SerializedName("genero") val genero: String,
    @SerializedName("foto")   val foto: String
)